Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/configuration.js

```diff
@@ -210,7 +211,7 @@ class Configuration {
             result += ' ' + this.style.split('-').map(word => word.charAt(0).toUpperCase() + word.slice(1)).join('-');
         } if (this.model != null)
             result += ' on ' + this.model;
-        if (this.architecture != null)
+        if (this.architecture != null && (!DEFAULT_ARCHITECTURE || this.architecture.search(DEFAULT_ARCHITECTURE) < 0))
```

Comment:
We should be consistent here and either rely on truthfulness or explicit comparisons to `null` (though the use of weak equality and not strict equality makes me sad).